timeseries/tests/unittests/models/test_sktime.py

```diff
@@ -126,32 +127,37 @@ def test_when_predict_called_with_test_data_then_predictor_inference_correct(
 
 
 @pytest.mark.parametrize("model_class", TESTABLE_MODELS)
+@pytest.mark.parametrize("hyperparameters", [{"seasonal_period": None}, {}])
 @pytest.mark.parametrize(
-    "freqstr, ts_length, expected_sp",
+    "freqstr, ts_length, expected_sp, should_warn",
     [
-        ("H", 100, 24),
-        ("2H", 100, 12),
-        ("B", 100, 5),
-        ("M", 100, 12),
-        ("H", 5, 1),
-        ("2H", 5, 1),
-        ("B", 5, 1),
-        ("M", 5, 1),
+        ("H", 100, 24, False),
+        ("2H", 100, 12, False),
+        ("B", 100, 5, False),
+        ("M", 100, 12, False),
+        ("H", 5, 1, True),
+        ("2H", 5, 1, True),
+        ("B", 5, 1, True),
+        ("M", 5, 1, True),
     ],
 )
-def test_when_fit_called_with_then_seasonality_period_set_correctly(
+def test_seasonal_period_is_inferred_correctly_when_set_to_none(
```

Comment:
we tried (although probably not always succeeded) to follow the [GWT](https://en.wikipedia.org/wiki/Given-When-Then) convention when naming test cases, sometimes omitting the G.